CMakeLists.txt

```
@@ -303,17 +306,17 @@ if(MINGW OR MSVC)
 
 	if(MSVC)
 		if(ENABLE_CCACHE)
-			# https://github.com/ccache/ccache/discussions/1154#discussioncomment-3611387
-			file(COPY_FILE
-				${CCACHE} ${CMAKE_BINARY_DIR}/cl.exe
-				ONLY_IF_DIFFERENT)
-
-			set(CMAKE_VS_GLOBALS
-				"CLToolExe=cl.exe"
-				"CLToolPath=${CMAKE_BINARY_DIR}"
-				"TrackFileAccess=false"
-				"UseMultiToolTask=true"
-			)
+			# https://github.com/mozilla/sccache
+			if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+			  string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
+			  string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
+			elseif(CMAKE_BUILD_TYPE STREQUAL "Release")
+			  string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
+			  string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
+			elseif(CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
+			  string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
+			  string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
```

can't you inject something like "disable /Zi and enable /Z7" to preset?

also, use tabs
```suggestion
				string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG}")
				string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG}")
			elseif(CMAKE_BUILD_TYPE STREQUAL "Release")
				string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE}")
				string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
			elseif(CMAKE_BUILD_TYPE STREQUAL "RelWithDebInfo")
				string(REPLACE "/Zi" "/Z7" CMAKE_CXX_FLAGS_RELWITHDEBINFO "${CMAKE_CXX_FLAGS_RELWITHDEBINFO}")
				string(REPLACE "/Zi" "/Z7" CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO}")
```

---

Newer cmake has option to do this automatically, but don't want to bump it.

Will fix, but only if this will actually help. So far I was unable to make msvc use ccache:
- ccache fails on running AUTO_RCC (old problem, unresolved)
- sccache seems to be doing nothing.

From what I found, switching from msbuild to ninja *should* fix ccache/sccache, but not idea how easy this change would be

---

switching generator should be trivial. In the worst case try `Ninja Multi-Config` in case msvc-specifics are tied to a multiconfig generator.

---

I read on various sources, switching to Ninja is solution, but also something about broken PDBs. Probably needs to be tested. 

Maybe worth to try this before switching to Ninja? https://github.com/ccache/ccache/discussions/1140#discussioncomment-3611405

---

Broken pdb's *should* be fixed by this code block in question.

But to be honest not much of a difference. I no longer have msvc set up to use pdb's, and Nullkiller who also used pdb is inactive. In other words - at the moment we can't open crash dumps, so pdb's are unused.

---

well, I have Windows with VS installed, can run it either normally or through a VM